Source/WebCore/rendering/svg/SVGVisitedRendererTracking.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#pragma once
+
+#include "RenderElement.h"
+#include <wtf/WeakHashSet.h>
+#include <wtf/WeakPtr.h>
+
+namespace WebCore {
+
+class SVGVisitedRendererTracking {
+public:
+    using VisitedSet = SingleThreadWeakHashSet<RenderElement>;
+
+    SVGVisitedRendererTracking(VisitedSet& visitedSet)
+        : m_visitedRenderers(visitedSet)
+    {
+    }
+
+    ~SVGVisitedRendererTracking() = default;
+
+    bool isEmpty() const { return m_visitedRenderers.isEmptyIgnoringNullReferences(); }
+    bool isVisiting(const RenderElement& renderer) { return m_visitedRenderers.contains(renderer); }
+
+    class Scope {
+    public:
+        Scope(SVGVisitedRendererTracking& tracking, const RenderElement& renderer, bool condition = true)
```

Comment:
Is the condition param still used?

Reply:
No, but it can indeed be helpful.

Reply:
I think it may be better to leave it out until it is needed.

Reply:
Ok, no big deal.